modules/hal_gigadevice/Kconfig

```diff
@@ -70,6 +70,11 @@ config GD32_LOW_SPEED_IRC_FREQUENCY
 	help
 	  Define value of low speed internal RC oscillator (IRC) in Hz
 
+config GD32_DBG_SUPPORT
+	bool "Use GD32 Debug features"
+	select USE_GD32_DBG
+	default y
```

Comment:
Isn't it possible to directly use `USE_GDB32_DBG`. This new symbol seems redundant to me.

Reply:
@carlocaione, @nandojve 

It can, but should better keep it to allow to turn off the debug function, I think.
A user is usually not allowed to configure the USE_... option in the modules/hal_* directory. (It seems to be an implicit rule.)
Adding GD32_DBG_SUPPORT is to enable/disable the debug function by the user.

Reply:
The _USE_foo_ and _HAS_foo_ symbols are not visible. The _USE_foo_ have the meaning to select something to build inside hal itself. I think there is no intention to have that kind of symbol inside zephyr mainline **code**. So, it seems to be redundant because, in general, zephyr drivers have been selecting the _USE_foo_ feature. In this case, GD32_DBG_SUPPORT is used to enable the hal_feature itself and can be used inside zephyr code. Unfortunately it is not possible to simple rely on the global DEBUG symbol.

@soburi, maybe it is important add documentation using `help` description for this config once it is used inside zephyr's code.

Reply:
@nandojve ,
Added `help` description about for this config.